can/interfaces/usb2can/serial_selector.py

```diff
@@ -38,10 +38,5 @@ def find_serial_devices(serial_matcher="ED"):
     objWMIService = win32com.client.Dispatch("WbemScripting.SWbemLocator")
     objSWbemServices = objWMIService.ConnectServer(".", "root\cimv2")
     items = objSWbemServices.ExecQuery("SELECT * FROM Win32_USBControllerDevice")
-    ids = map(lambda item: item.Dependent, items)
-
-    return [
-        string[len(string) - 9:len(string) - 1]
-        for string in ids
-        if serial_matcher in string
-    ]
+    ids = (item.Dependent[-8:] for item in items)
```

Comment:
This needs to be `[-9:]` to match the last eight bytes.

Reply:
No, this should be correct:

```python
>>> "..12345678"[-8:] == "12345678"
>>> True
```

I just tested it on Python 2 & 3.

Reply:
Strange, maybe the win32 returned string has some trailing whitespace then? It only returned "D000200" and other 7-character entries in my testing, evidenced by a print statement after the list comprehension. So it failed to match the adapter because of the missing E. Using - 9 fixed it. 

I will check again tomorrow when back at work.

Reply:
There is a trailing quote on the strings returned by WMI. See the dump of `ids` below. Should probably use `str.strip()` to remove them.
```
['777AF&0"',
 '3E7&0&1"',
 'AC0&0&7"',
 '&0&0000"',
 '&0&0000"',
 '&0&0001"',
 '&0&0000"',
 '&0&0001"',
 'AC0&0&8"',
 '&0&0000"',
 '3F4F5&0"',
 'B7C&0&1"',
 'CA53C&0"',
 '2E3&0&1"',
 'D000200"',
 '3646476"',
 '&0&0000"',
 '46476&0"',
 '46476&1"',
 '46476&2"',
 '46476&3"',
 '&0&0001"',
 '&0&0000"',
 'L00INBZ"',
 'A\\\\0000"',
 '2E3&0&2"',
 '&0&0000"']
```

Reply:
I just tested with `.strip('\"')`, and it correctly builds the `ids` list. However, the function still returns an empty list, although I checked that `.starts_with("ED")` actually does return `True`.

What's going on here?!

Reply:
Okay, never mind. My print statement exhausted the generator expression `ids` so the actual list comprehension for the returned result could not yield any more results... Python yaaay!